Cleanup

krystian-hebel opened the pull request.

krystian-hebel committed:
documentation/*CI*: remove documents

CI isn't working for quite some time already. Having those section in
the documentation only caused confusion among users.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
treewide: use relative paths to images

This reduces number of lines printed by mkdocs, which should make
catching real issues easier.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
mkdocs.yml: unwrap submenus

This both removes info about absolute links, as well as allows for
setting order of entries manually (it used to be sorted alphabetically).
In addition, soecifications and steering committee menus had one empty
submenu level each, those have been removed.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
mkdocs.yml: reorder entries, create more categories

No changes to content, other than updating the links on the home page.
Suggestions for new pages are in comments in mkdocs.yml.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
docs/references: add to nav menu

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
blueprints/*_Grub_Late_Launch.md: represent implemented state

This also fixes formatting of deployed page.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
blueprints/AMD_Secure_Kernel_Loader.md: rename from LZ, update

Changes include addition of SLRT, generic name for kernel configuration
and reworked build option (BITS).

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
_config.yml: remove remaining Grayscale file

This is not used in Mkdocs approach.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
CONTRIBUTING.md: remove file

The same file exists in docs/dev-docs, from which it is used on the
website. It is identical, except few typos that were fixed in
docs/dev-docs. In addition, one whitespace issues were fixed in the
remaining file.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
dev-docs/Late_Launch_Overview.md: reword SKINIT section, formatting fixes

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
theory/Architecture.md: rewrap

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
user-docs/requirements.md: add file

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
user-docs/install_aem.md: add instructions

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
docs/index.md: add link to user documentation on home page

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

philipanda committed:
docs/events.md: Add new conferences & events since 2020

The slides for FOSDEM 2021 are missing.

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
blueprints/AMD_Landing_Zone, coreboot_DRTM: move from documentation repo

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
docs/references: Update documents to newest revisions

secure-launch-specification.rst: move from documentation repo
UseCases.md: typo
Late_Launch_Overview.md: typo
FAQ.md: linewrap
AMD_Landing_Zone.md: Remove, replaced with SKL

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

m-iwanicki committed:
pre-commit: add pre-commit with default hooks + lychee

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
.gitignore: add 'venv/' to .gitignore

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
requirements.txt: freeze and clean requirements

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
README.md: Add minimal readme with local preview info

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
mkdocs.yml: remove markdown-include, add redirects, freeze awesome-pages

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

Commit:
LICENSE.md: format per markdownlint-fix

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

Commit:
events.md: Remove the 'Video' bulletpoint

The video for Platform Security Summit 2019 was never published, so the
bulletpoint mentioning the video was removed.

SergiiDmytruk committed:
theory/glossary: add acronyms to couple entries

Makes it easier to look up acronyms in the glossary.

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
theory/glossary: add thirteen more terms

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
blueprints/Xen_Late_Launch.md: add

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
blueprints/SLRT.md: add

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
blueprints/index.md: add more details

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

philipanda committed:
FAQ.md: Add reformatted Background.md contents

Originally submitted to https://github.com/TrenchBoot/documentation/pull/11

Co-authored-by: Garnet Grimm
Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
user-docs/troubleshooting.md: add

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
user-docs/build_grub.md: add

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

philipanda committed:
docs/QUICKSTART.md: Add from documentation repository

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
UseCases.md: Add links to blogs, publications and a book

m-iwanicki committed:
README.md: add short pre-commit information

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

SergiiDmytruk committed:
blueprints/index.md: fix links

References are case-insensitive, so [Linux] and [linux] conflict and the
latter wins.

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

krystian-hebel committed:
assets/style.css: make the text easier for the eyes

Larger blocks of white text on black background were hard to read. Reduce
the contrast by making the text grey for the body. Headers and links are
left as they were.

This commit also cleans up the CSS a bit by moving variables to global
scope and unifying spelling of grey/gray.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

m-iwanicki committed:
pre-commit: fix pre-commit failures

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
README.md: add mkdocs output verification

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

philipanda committed:
UseCases.md: Add a note that publication is in german

Co-authored-by: Krystian Hebel <krystian.hebel@3mdeb.com>

philipanda committed:
FAQ.md: remove outdated info, fix factual errors

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
docs/publications.md: Move publications to seperate file

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
docs/QUICKSTART.md: Remove troubleshooting

it was moved to its own page

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
blueprints/AMD_coreboot_DRTM_payload.md: Add to the website

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
user-docs/QUICKSTART.md: Update according to documentation repo

Co-authored-by: Ross Philipson
Co-authored-by: Daniel P. Smith
Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
mkdocs.yml: Move quickstart guide lower in the nav bar

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

krystian-hebel commented:
Last force push fixed link to `Paving the Path for TrenchBoot DRTM in Xen Hypervisor` added in https://github.com/TrenchBoot/TrenchBoot.github.io/pull/32 that was both malformed and wrong. It also squashed some commits related to that and next PR to avoid reverting and adding the same change later.

SergiiDmytruk committed:
user-docs/build_linux.md: add

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

m-iwanicki committed:
pre-commit: add MD053 to .markdownlint.yaml

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

krystian-hebel committed:
dev-docs/CONTRIBUTING.md: describe branch naming convention

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

SergiiDmytruk committed:
dev-docs/CONTRIBUTING.md: better formatting of "License" section

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

macpijan commented:
A lot of work has been put into this, is there anything blocking the current cleanup from merging? @krystian-hebel @SergiiDmytruk ?

I realize there are still some boxes to check in: https://github.com/TrenchBoot/trenchboot-issues/issues/29
But perhaps the current state of this branch is already superior to the existing state of documentation website?

I wanted to take advantage of these improvements and point someone to the updated events page, but it is still not hosted in the recent version.

philipanda committed:
FAQ.md: Language and interpunction

Edited-by: Amanda Ferguson <af.jenks@gmail.com>
Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
FAQ.md: Explain why TB uses an intermediate loader, fix typos

Co-authored-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

krystian-hebel committed:
docs: normalize TrenchBoot spelling

The only 'Trenchboot' left is that in matrix room link, it must remain
that way.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

rossphilipson reviewed (approved):
The changes LGTM and once this is in we can more easily read everything and fix up anything that still needs attention.

Reviewed-by: Ross Philipson <ross.philipson@oracle.com>
